src/lib/popover/popover.component.ts

```diff
@@ -223,48 +259,54 @@ export class SatPopover implements OnInit, OnDestroy {
 
   constructor(
     private _focusTrapFactory: FocusTrapFactory,
+    private _anchoringService: SatPopoverAnchoringService,
+    private _viewContainerRef: ViewContainerRef,
     @Optional() @Inject(DOCUMENT) private _document: any
   ) { }
 
   ngOnInit() {
     this._setAlignmentClasses();
   }
 
-  ngOnDestroy() {
-    if (this._notifications) {
-      this._notifications.dispose();
-    }
-  }
-
   /** Open this popover. */
   open(options: SatPopoverOpenOptions = {}): void {
-    const notification = new PopoverNotification(NotificationAction.OPEN, options);
-    this._dispatchActionNotification(notification);
+    if (this._anchor) {
+      this._anchoringService.openPopover(options);
+      return;
+    }
+
+    throw getUnanchoredPopoverError();
   }
 
   /** Close this popover. */
   close(value?: any): void {
-    const notification = new PopoverNotification(NotificationAction.CLOSE, value);
-    this._dispatchActionNotification(notification);
+    this._anchoringService.closePopover(value);
   }
 
   /** Toggle this popover open or closed. */
   toggle(): void {
-    const notification = new PopoverNotification(NotificationAction.TOGGLE);
-    this._dispatchActionNotification(notification);
+    this._anchoringService.togglePopover();
   }
 
   /** Realign the popover to the anchor. */
   realign(): void {
-    const notification = new PopoverNotification(NotificationAction.REALIGN);
-    this._dispatchActionNotification(notification);
+    this._anchoringService.realignPopoverToAnchor();
   }
 
   /** Gets whether the popover is presently open. */
   isOpen(): boolean {
     return this._open;
   }
 
+  /** Allows programmatically setting a custom anchor. */
+  setCustomAnchor(
+    viewContainer: ViewContainerRef,
+    el: ElementRef<HTMLElement> | HTMLElement
+  ): void {
```

Comment:
One of the tests allowed programmatically setting a custom anchor element. Given that the SatPopover now allows setting the anchor via its `anchor` property, its not clear to me that this functionality is still needed. For the time being, I've kept the functionality by providing this method (which is the only way to programmatically set the `ViewContainerRef` associated with the popover.